moveit_ros/planning/planning_components_tools/src/compare_collision_speed_checking_fcl_bullet.cpp

```diff
@@ -0,0 +1,355 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Jens Petit
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jens Petit */
```

Comment:
Does this file belong in the moveit_tutorials instead? as example code?

Reply:
I am not sure, also not happy about the current location of the file. But tutorial might also be hard to justify.. any other ideas? I placed it there because there is a similar script in the folder [here](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning/planning_components_tools/src/evaluate_collision_checking_speed.cpp). I assume it was done because they wanted to check the speed directly via the planning scene and therefore it was natural to be put into `moveit_ros`.

Reply:
IMO benchmarks aren't tutorials, and you shouldn't have to go through the tutorial to get benchmark speeds for which collision checker you want. I'm not sure it's the best for discoverability, but I really like how Mark did things in the cached ik plugin: https://github.com/ros-planning/moveit/tree/master/moveit_kinematics/cached_ik_kinematics_plugin#measuring-ik-solver-performance. 

Maybe what you're thinking of (picture based tutorial) could go in a README in the Bullet collision checker folder, and just reference this script?

Reply:
But isn't it strange if the script is in a totally different folder than the the README?

Reply:
Yeah, it is, but no one is going to have just `moveit_core` installed and not `moveit_ros`. Ideally, this script doesn't need to use the planning scene monitor. I think it was originally set up that way so you could use any scene that you loaded independently.  If you have the time you could refactor the script to just directly load a robot from moveit resources, but I have a feeling that would get complicated quickly so I wouldn't recommend it.

Reply:
But isn't this just example code for benchmarking a particular robot? I've been slowly migrating all example code to the tutorials, and adding documentation along side it for centralized location for learning moveit.

but this isn't a blocker for me.